src/common/utils/http.ts

```diff
@@ -0,0 +1,48 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export async function httpGet(url: string, options?: request.CoreOptions): Promise<string> {
+
+    const store = container.get("store") as Store<RootState>;
+    const locale = store.getState().i18n.locale;
+    const headers = Object.assign(
+        {},
+        options.headers,
+        {
+            "User-Agent": "readium-desktop",
+            "Accept-Language": `${locale},en-US;q=0.7,en;q=0.5`,
```

Comment:
Strictly-speaking this is problematic, so I must flag it up (however, note that in practice the `readium-desktop` code is unlikely to trigger the edge-case):

Callers of the function `httpGet()` can pass `options.headers` = `{ "user-agent":  "MyOverriddenUserAgent" }` (note the lower-case key ... HTTP headers are case-insensitive). This will result in the concatenation of `"readium-desktop"` and `"MyOverriddenUserAgent"`, because the `request` lib utilises `caseless` in order to ignore case variants (see https://github.com/request/request/blob/212570b6971a732b8dd9f3c73354bcdda158a737/request.js#L159 and https://github.com/request/caseless/blob/98079105e15e04308d1f3becc3e8efab8ddd7f0e/index.js#L59 ).

One solution would be to force-lower-case the object/dictionary keys of a `options.headers` clone, and to use lower-case `"user-agent"` and `"accept-language"` in order to ensure this takes precedence in `Object.assign()`.

Reply:
done